cloudinit/sources/DataSourceLXD.py

```diff
@@ -64,19 +55,46 @@ def generate_fallback_network_config() -> dict:
                 " Rendering default network config.",
                 err,
             )
-            return network_v1
+            return default_name
         if virt_type.strip() in (
             "kvm",
             "qemu",
         ):  # instance.type VIRTUAL-MACHINE
             arch = util.system_info()["uname"][4]
             if arch == "ppc64le":
-                network_v1["config"][0]["name"] = "enp0s5"
+                return "enp0s5"
             elif arch == "s390x":
-                network_v1["config"][0]["name"] = "enc9"
+                return "enc9"
             else:
-                network_v1["config"][0]["name"] = "enp5s0"
-    return network_v1
+                return "enp5s0"
+    return default_name
+
+
+def generate_network_config(
+    nics: Optional[List[str]] = None,
+) -> Dict[str, Any]:
+    """Return network config V1 dict representing instance network config."""
+    if not nics:
+        primary_nic = _get_fallback_interface_name()
+    elif len(nics) > 1:
+        fallback_nic = find_fallback_nic()
+        if not fallback_nic or fallback_nic not in nics:
+            primary_nic = nics[0]
+        else:
+            primary_nic = fallback_nic
```

Comment:
Would this be the right place to raise awareness about needing to set `user.network-config` or `cloud-init.network-config` and cloud-init only setting up network on primary_nic?
```suggestion
            primary_nic = fallback_nic
            LOG.warning("Found multiple NICs at lxd/1.0/devices and no applicable `user.network-config` or `cloud-init.network-config` set. Only configuring DHCP on %s", primary_nic)
```

Reply:
I don't think so. I'm guessing the more common LXD use case will be configuring it yourself.

Reply:
We decided in standup: it's best really to add documentation for hotplug enablement of LXD datasource informing the user that desired network config should be set via `lxc config set user.network-config` or `cloud-init.network-config` prior to attaching/detaching multiple NICS. And absence of lxc network-config will result in dhcp on primary NIC.